google/cloud/storagecontrol/v2/samples/storage_control_folder_samples.cc

```diff
@@ -0,0 +1,179 @@
+      std::string const& bucket_name,
+      std::string const& folder_id) {
+    auto const name = std::string{"projects/_/buckets/"} + bucket_name + std::string{"/folders/"} + folder_id;
+    auto status = client.GetFolder(name);
+    if (!status) throw std::move(status).status();
+
+    std::cout << "Got folder: " << status->name() << "\n";
+  }
+  // [END storage_control_get_folder]
+  (std::move(client), argv.at(0), argv.at(1));
+}
+
+void ListFolders(::google::cloud::storagecontrol_v2::StorageControlClient client,
+                  std::vector<std::string> const& argv) {
+  // [START storage_control_list_folders]
+  namespace storagecontrol = ::google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+      std::string const& bucket_name) {
+    auto const parent = std::string{"projects/_/buckets/"} + bucket_name;
+    auto stream_range = client.ListFolders(parent);
+    for(auto status : stream_range) {
+      if (!status) throw std::move(status).status();
+      std::cout << status->name() << "\n";
+    }
+
+    std::cout << bucket_name << std::endl;
+  }
+  // [END storage_control_list_folders]
+  (std::move(client), argv.at(0));
+}
+
+void RenameFolder(::google::cloud::storagecontrol_v2::StorageControlClient client,
+                  std::vector<std::string> const& argv) {
+  // [START storage_control_rename_folder]
+  namespace storagecontrol = ::google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+      std::string const& bucket_name,
+      std::string const& source_folder_id,
+      std::string const& dest_folder_id) {
+
+    auto name = std::string{"projects/_/buckets/"} + bucket_name + std::string{"/folders/"} + source_folder_id;
+    auto status_future = client.RenameFolder(name, dest_folder_id);
+    auto status = status_future.get();
+    if (!status) throw std::move(status).status();
+
+    std::cout << "Renamed: " << source_folder_id << " to: " << dest_folder_id << "\n";
+  }
+  // [END storage_control_rename_folder]
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2));
+}
+
+void AutoRun(::google::cloud::storagecontrol_v2::StorageControlClient client,
+              std::vector<std::string> const& argv) {
+    namespace examples = ::google::cloud::testing_util;
+    namespace storagecontrol = ::google::cloud::storagecontrol_v2;
+    if (argv.size() != 1) throw examples::Usage{"auto"};
+    [](storagecontrol::StorageControlClient client,
+        std::string const& bucket_name) {
+      auto generator = google::cloud::internal::DefaultPRNG(std::random_device{}());
+      auto const prefix = "storage-control-samples";
+      auto const folder_id =
+        prefix + std::string{"-"} + google::cloud::internal::Sample(generator, 32, "abcdefghijklmnopqrstuvwxyz");
+      auto const dest_folder_id = 
+        prefix + std::string{"-"} + google::cloud::internal::Sample(generator, 32, "abcdefghijklmnopqrstuvwxyz");
+
```

Comment:
Consider a "cleanup" call here, to garbage collect any folders leaked from previous runs.  We would typically do something like:

https://github.com/googleapis/google-cloud-cpp/blob/1adeea2ce218aa9f874db45929e144ae3fb38c97/google/cloud/storage/testing/remove_stale_buckets.cc#L47-L57

Reply:
Added a cleanup call, working through adding a test for it.

Reply:
A test would be nice, but feel free to do that in a future PR.  I did not mean to block this PR on having testable cleanups.